### ISSUE TITLE
CS8509 is not trigged when null is missing as a pattern when matching a nullable value type

### ISSUE THREAD

**Issue by NiekGr** [dotnet/roslyn#38909]
Dear Sir/Madam,

**Version Used**: 
Visual Studio 2019 16.3.1

**Steps to Reproduce**:

1. Use a switch expression on a nullable value type
2. Do not use the `null` pattern
```
bool? input = null;
input switch // No warning
{
    true => "true",
    false => "false"
};
```

**Expected Behavior**:

The expected behaviour is a warning forunexhaustive patterns, CS8509.
Just as in the following cases
```
double? input = null;
input switch // Warning	CS8509 The switch expression does not handle all possible inputs (it is not exhaustive).
{
    true => "true",
    null => "null"
};

#nullable enable
string? input = null;
input switch // Warning	CS8509 The switch expression does not handle all possible inputs (it is not exhaustive).
{
    { } => "not null",
};
```

**Actual Behavior**:
No warning while there is an unexhaustive pattern.

Thank you in advance.

Regards,
NiekGr

**Comment by gafter** [comment]
We require `#nullable enable` in order for the compiler to track whether or not possible null values are handled.  Please try that.

**Comment by NiekGr** [comment]
Sorry I had totally missed your reaction.

Nullable reference types is a new feature in C# 8, but nullable value types are available since C# 2. I would expect the exhaustive patterns checker to support nullable value types without enabling nullable reference types.

**Comment by gafter** [comment]
Diagnostics relating to possibly null values is a new feature in C# 8, and is enabled by the `#nullable enable` preprocessor directive.  We perform the analysis of where null values can reach only when that is enabled, even for `Nullable<T>` value types.  That is by design.
